web/jsonapi/data.go

```diff
@@ -0,0 +1,85 @@
+package jsonapi
+
+import (
+	"encoding/json"
+
+	"github.com/cozy/cozy-stack/couchdb"
+)
+
+// Object is an interface to serialize something to a JSON-API Object
+type Object interface {
+	couchdb.Doc
+	SelfLink() string
+	Relationships() RelationshipMap
+	Included() []Object
+}
```

Comment:
Well, in fact, it looks like `struct`s don't implement the `json.Marshaler` interface, even if we can serialize them to JSON.